### PR TITLE
feat(pocket): adding a dataset_metadata.yaml file to grant default access to the pocket dataset

### PR DESCRIPTION
**Pull request by braunreyes**
Prefect needs this access along with Mozilla Confidential

Referenced https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/activity_stream/dataset_metadata.yaml as an example

**[SOCIALPLAT-412](https://mozilla-hub.atlassian.net/browse/SOCIALPLAT-412)**

Checklist for reviewer:

- [X] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [X] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [N/A] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [N/A] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [N/A] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

[SOCIALPLAT-412]: https://mozilla-hub.atlassian.net/browse/SOCIALPLAT-412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by whd — COMMENTED** [review]
Apologies for the somewhat lengthy commentary but there's a fair amount of context needed when dealing with access patterns and metadata changes.

First: this access was the discussion of https://mozilla-hub.atlassian.net/browse/DSRE-1327. In that ticket Mark Reid was determined to be the default owner of the data and therefore the approver of this access change. With increased Pocket integration into the Data Platform I hope we can shift ownership to folks with more active context into the data so that decisions like this can be made more quickly. At any rate, there is no policy issue with this PR since it has been approved, so that's a great start.

The primary interface for consuming `dataset_metadata.yaml` from bigquery-etl is the `generated-sql` branch. You can see from that branch that in cases where `dataset_metadata.yaml` is not specified it gets auto-created in CI e.g. https://github.com/mozilla/bigquery-etl/blob/generated-sql/sql/moz-fx-data-shared-prod/pocket/dataset_metadata.yaml. For consistency, you should use this file as the basis for your `dataset_metadata.yaml` file, and simply add `workgroup:pocket/prefect` to `workgroup_access`. Your PR as filed has some slight discrepancies with the generated content.

https://docs.telemetry.mozilla.org/cookbooks/bigquery/querying.html#table-layout-and-naming has some description of the dataset layout of our platform. A consequence of the layout is that granting access to the `view` dataset that this PR is doing is not usually sufficient since access to the underlying tables will not be granted. You should add to this PR another `dataset_metadata.yaml` file for [pocket_derived](https://github.com/mozilla/bigquery-etl/blob/generated-sql/sql/moz-fx-data-shared-prod/pocket_derived/dataset_metadata.yaml) that also grants `workgroup:pocket/prefect` access in order for this PR to grant the relevant access. 

Nonstandard dataset access patterns like this currently require a companion PR to be filed in the ops repo. I don't know if you have access to this repo yet but I've filed https://github.com/mozilla-services/cloudops-infra/pull/5009 to pair with this one to keep the metadata synced. This synchronization is only necessary for dataset metadata; see below.

From https://mozilla-hub.atlassian.net/browse/SOCIALPLAT-412 it seems you're concerned with access to a specific view: `pocket.tile_spoc_ids`. In the spirit of [PoLP](https://mozilla-hub.atlassian.net/wiki/spaces/SRE/pages/27924789/Data+Access+Workgroups#DataAccessWorkgroups-Workgroupaccessforautomationandapplicationserviceaccounts) an alternative mode of access here would be to add table-level `workgroup_access` to https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml#L5 and https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/pocket_derived/spoc_tile_ids_v1/metadata.yaml#L11. In this case since table ACLs are additive to dataset ones you would only need to specify:
```yaml
workgroup_access:
- role: roles/bigquery.dataViewer
  members:
  - workgroup:pocket/prefect
```
since `workgroup:mozilla-confidential` access is granted at the dataset level (see e.g. [here](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/activity_stream_bi/impression_stats_flat_v1/metadata.yaml#L24)). I recommend using table ACLs if you're confident that `workgroup:pocket/prefect` only needs granular access to some specific tables and no others. If you're expecting it to need access to more tables within these datasets over time then using `dataset_metadata.yaml` might be a better fit.

Lastly CI won't work for forks of this PR, by design. Reviewers can trigger CI once the PR is ready, but in this case I may not bother since I have a strong understanding of the particulars here and there isn't a lot of CI covering cases related to dataset metadata files.

**Comment by braunreyes** [comment]
@whd I went ahead and made the additive changes you suggested.  Thank you so much for providing that explanation for how the grants work.  this was very helpful

File: sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml [hunk 1]
```diff
@@ -3,3 +3,7 @@ description: |-
   Lookup table of the currently registered tile IDs for Pocket sponsored content
 owners:
   - jklukas@mozilla.com
+workgroup_access:
+- role: roles/bigquery.dataViewer
+  members:
+  - workgroup:pocket/prefect
```
Comment (whd) [inline sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml:9]:
```suggestion
  - workgroup:pocket/prefect
```

File: sql/moz-fx-data-shared-prod/pocket_derived/spoc_tile_ids_v1/metadata.yaml [hunk 2]
```diff
@@ -9,3 +9,7 @@ scheduling:
   dag_name: bqetl_pocket
   date_partition_parameter: null
   parameters: ["submission_date:DATE:{{ds}}"]
+workgroup_access:
+- role: roles/bigquery.dataViewer
+  members:
+  - workgroup:pocket/prefect
```
Comment (whd) [inline sql/moz-fx-data-shared-prod/pocket_derived/spoc_tile_ids_v1/metadata.yaml:15]:
```suggestion
  - workgroup:pocket/prefect
```

File: sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml (hunk 1 above)
Comment (whd) [reply sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml:9]: I think the yaml lint is failing because of missing newlines at EOF.

Comment (braunreyes) [reply sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml]: added the missing newlines

File: sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml [hunk 3]
```diff
@@ -3,3 +3,7 @@ description: |-
   Lookup table of the currently registered tile IDs for Pocket sponsored content
 owners:
   - jklukas@mozilla.com
+workgroup_access:
```
Comment (whd) [inline sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml:6]: Apologies for not looking at the CI more closely. It appears the lint failure is as follows:

```
./sql/moz-fx-data-shared-prod/pocket/spoc_tile_ids/metadata.yaml
  7:1       error    wrong indentation: expected 2 but found 0  (indentation)
  9:3       error    wrong indentation: expected 4 but found 2  (indentation)

./sql/moz-fx-data-shared-prod/pocket_derived/spoc_tile_ids_v1/metadata.yaml
  13:1      error    wrong indentation: expected 2 but found 0  (indentation)
  15:3      error    wrong indentation: expected 4 but found 2  (indentation)
```

So the indentation needs to be:
```yaml
workgroup_access:
  - role: roles/bigquery.dataViewer
    members:
      - workgroup:pocket/prefect
```

**Review by whd — APPROVED** [review]
r+wc

**Pull request merged**